addons/account/models/account_move.py

```diff
@@ -2048,6 +2049,112 @@ def _validate_taxes_country(self):
+        else:  # In case of an entry
+            return super()._get_product_catalog_domain()
+
+    def _default_order_line_values(self):
+        default_data = super()._default_order_line_values()
+        new_default_data = self.env['account.move.line']._get_product_catalog_lines_data()
+        return {**default_data, **new_default_data}
+
+    def _get_product_catalog_order_data(self, products, **kwargs):
+        return {product.id: self._get_product_price_and_data(product) for product in products}
+
+    def _get_product_price_and_data(self, product):
+        """
+            This function will return a dict containing the price of the product. If the product is a sale document then
+            we return the list price (which is the "Sales Price" in a product) otherwise we return the standard_price
+            (which is the "Cost" in a product).
+            In case of a purchase document, it's possible that we have special price for certain partner.
+            We will check the sellers set on the product and update the price and min_qty for it if needed.
+        """
+        self.ensure_one()
+        product_infos = {'price': product.list_price if self.is_sale_document() else product.standard_price}
+
+        # Check if there is a price and a minimum quantity for the order's vendor.
+        if self.is_purchase_document() and self.partner_id:
+            seller = product._select_seller(
+                partner_id=self.partner_id,
+                quantity=None,
+                date=self.invoice_date,
+                uom_id=product.uom_id,
+                ordered_by='min_qty',
+                params={'order_id': self}
+            )
+            if seller:
+                product_infos.update(
+                    price=seller.price,
+                    min_qty=seller.min_qty,
+                )
+        return product_infos
+
+    def _get_product_catalog_record_lines(self, product_ids):
+        grouped_lines = defaultdict(lambda: self.env['account.move.line'])
+        for line in self.line_ids:
+            if line.display_type == 'product' and line.product_id.id in product_ids:
+                grouped_lines[line.product_id] |= line
+        return grouped_lines
+
+    def _update_order_line_info(self, product_id, quantity, **kwargs):
+        """ Update account_move_line information for a given product or create a
+        new one if none exists yet.
+        :param int product_id: The product, as a `product.product` id.
+        :param int quantity: The quantity selected in the catalog
+        :return: The unit price of the product, based on the pricelist of the
+                 sale order and the quantity selected.
+        :rtype: float
+        """
+        move_line = self.line_ids.filtered(lambda line: line.product_id.id == product_id)
+        if move_line:
+            if quantity != 0:
+                move_line.quantity = quantity
+            elif self.state in {'draft', 'sent'}:
```

Comment:
Yop, I'm checking my linter and see that this `sent` state doesn't exist.
Should I just remove of check `is_move_sent` instead? :thinking:

Reply:
I think it's a bad copy paste form sale_order :eyes: Since we don't have that on move maybe we can just put self.state == draft